packages/Webkul/Admin/src/DataGrids/CategoryProductDataGrid.php

```diff
@@ -195,5 +242,37 @@ public function prepareActions()
                 return true;
             },
         ]);
+
+        $this->addAction([
+            'title'        => trans('admin::app.datagrid.delete'),
+            'method'       => 'POST',
+            'route'        => 'admin.catalog.products.delete',
+            'confirm_text' => trans('ui::app.datagrid.mass-action.delete', ['resource' => 'product']),
+            'icon'         => 'icon trash-icon',
+        ]);
+
+        $this->addAction([
+            'title'  => trans('admin::app.datagrid.copy'),
+            'method' => 'GET',
+            'route'  => 'admin.catalog.products.copy',
+            'icon'   => 'icon copy-icon',
+        ]);
+    }
+
+    /**
+     * Render quantity view.
+     *
+     * @param  object  $row
+     * @return \Illuminate\Contracts\View\View|\Illuminate\Contracts\View\Factory
+     */
+    private function renderQuantityView($row)
+    {
+        $product = $this->productRepository->find($row->product_id);
+
+        $inventorySources = $this->inventorySourceRepository->findWhere(['status' => 1]);
+
+        $totalQuantity = $row->quantity;
+
+        return view('admin::catalog.products.datagrid.quantity', compact('product', 'inventorySources', 'totalQuantity'))->render();
```

Comment:
When I added this to the products table, I was not aware of the consequences but now I think we should not add this.

Because this will hit multiple queries in the database. For e.g. if I select 50 from the pagination then it will hit the 50 queries in the database.

Try to optimize this if possible.